lib/portlayer/storage/image_cache.go

```diff
@@ -60,9 +60,20 @@ func NewLookupCache(ds ImageStorer) *NameLookupCache {
 	}
 }
 
+// isRetry will check the error for retryability - if so reset the cache
+func (c *NameLookupCache) isRetry(op trace.Operation, err error) bool {
+	if tasks.IsRetryError(op, err) {
+		op.Debugf("%s is retryable, resetting store cache", err)
+		c.storeCache = make(map[url.URL]*index.Index)
```

Comment:
Am I correct in assuming that there's no single place where we can purge the cache before regenerating it, e.g at the start of GetImageStore, and that's why it's done in the error type check?

Reply:
I didn't see a good place to do it -- originally I had it as part of the retry function, but it didn't quite work....so went with this approach...open to suggestions.